monai/apps/pathology/transforms/spatial/array.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Tuple, Union
+
+import torch
+
+from monai.transforms.transform import Transform
+
+__all__ = ["SplitOnGrid"]
+
+
+class SplitOnGrid(Transform):
+    """
+    Split the image into patches based on the provided grid shape.
+    This transform works only with torch.Tensor inputs.
+
+    Args:
+        grid_shape: a tuple or an integer define the shape of the grid upon which to extract patches.
+            If it's an integer, th evalue will be repeated for each dimension. Default is 2x2
+        patch_size: a tuple or an integer that defines the output patch sizes.
+            If it's an integer, the value will be repeated for each dimension.
+            If None (default), the patch size will be infered from the grid shape.
+
+    Note: the shape of the input image is infered based on the first image used.
```

Comment:
Should we `prepare_params()` for every image? If so, we don't need `ready` flag?

Reply:
It's only for the this first image to avoid unnecessary computation for each one.